tests/prague/eip663_dupn_swapn_exchange/test_dupn.py

```diff
@@ -0,0 +1,57 @@
+"""
+abstract: Tests [EIP-663: SWAPN, DUPN and EXCHANGE instructions](https://eips.ethereum.org/EIPS/eip-663)
+    Tests for the DUPN instruction.
+"""  # noqa: E501
+
+import pytest
+
+from ethereum_test_tools import Account, Environment, StateTestFiller, TestAddress, Transaction
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.eof.v1.constants import NON_RETURNING_SECTION
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from ..eip3540_eof_v1.spec import EOF_FORK_NAME
+from . import REFERENCE_SPEC_GIT_PATH, REFERENCE_SPEC_VERSION, tx
+
+REFERENCE_SPEC_GIT_PATH = REFERENCE_SPEC_GIT_PATH
+REFERENCE_SPEC_VERSION = REFERENCE_SPEC_VERSION
+
+
+@pytest.mark.valid_from(EOF_FORK_NAME)
+def test_dupn_all_valid_immediates(
+    tx: Transaction,
+    state_test: StateTestFiller,
+):
+    """
+    Test case for all valid DUPN immediates.
+    """
+
+    n = 256
+    values = range(0xD00, 0xD00 + n)
+
+    eof_code = Container(
+        sections=[
+            Section.Code(
+                code=b"".join(Op.PUSH2(v) for v in values)
+                + b"".join(Op.DUPN(x) + Op.PUSH2(x) + Op.SSTORE for x in range(0, n))
```

Comment:
```suggestion
                + b"".join(Op.SSTORE(x, Op.DUPN[x]) for x in range(0, n))
```
There was a slight semantics change recently where we added square-brackets to the opcodes that have a data portion, so this should work as well.
Right now both `Op.DUPN(x)` and `Op.DUPN[x]` result in the exact same bytecode, but for others like `RJUMPV` it makes it much more readable, so we are trying to push for the new semantics.

Reply:
Somehow, this expression doesn't work if changed `(v)` to `[v]`: `b"".join(Op.PUSH2(v) for v in values)`.

Reply:
Somehow, this expression doesn't work if changed `(v)` to `[v]`: `b"".join(Op.PUSH2(v) for v in values)`.

Reply:
Yes, I tried to find a workaround but sadly there does not seem to be any straight-forward one because the resulting `Opcode` type from the subscript operation is not detected as something that can be bytes-concatenated.

For the time being just leave it as is:
```python
b"".join(Op.PUSH2(v) for v in values)
```